test/e2e/validation_test.go

```diff
@@ -165,37 +154,47 @@ func validateInstanceSecret(kubeclient kubernetes.Interface, namespace, instance
 			return false, err
 		}
 		encodedCreds := instanceSecret.Data[instanceID]
-
-		var creds tracker.Credentials
 		if err := json.Unmarshal(encodedCreds, &creds); err != nil {
 			return false, errors.Wrap(err, "unmarshalling creds failed")
 		}
-
-		if err := validateConnectivity(ipAddress, instanceID, creds); err == nil {
-			log.Println("Successfully validated the SSH Connection")
-			return true, nil
-		}
-
-		log.Printf("failed with error for creds %v: %v", creds, err)
-		return false, nil
+		return true, nil
 	})
+	return creds, err
+}
+
+// validateInstanceSecret validates the instance secret.
+func (tc *testContext) validateInstanceSecret(instanceID string) error {
+	ipAddress, err := tc.getInstanceIP(instanceID)
+	if err != nil {
+		return err
+	}
+	creds, err := tc.getCredsFromSecret(instanceID)
+	if err != nil {
+		return err
+	}
+	if creds == (tracker.Credentials{}) {
+		return errors.New("expected credentials to be present but got a nil value")
+	}
+	windowsVM, err := tc.getWindowsVM(ipAddress, instanceID, creds)
+	if err != nil {
+		return err
+	}
+	gc.windowsVMs = append(gc.windowsVMs, windowsVM)
```

Comment:
It'll be used by @aravindhp in #21

Reply:
In that case shouldnt he add it in his PR?
Genuine question.

Reply:
We are just trying to avoid code churn here. But if you insist we can remove this from here and I can add it in my PR.

Reply:
Im not insisting. If its for sure being used thats fine. I was under the impression its best practice  in order to avoid the situation where we think it will be used and then not actually used, resulting in it being left around for no reason.